### PR TITLE
feat: Add `precheck` to Milvus connector

### PR DESCRIPTION
**Pull request by ds-filipknefel**
Add `precheck` method to Milvus connector together with tests verifying that it will succeed on an existent collection and fail on a nonexistent one.

### CODE REVIEW

File: unstructured_ingest/v2/processes/connectors/milvus.py [hunk 1]
```diff
@@ -174,6 +173,14 @@ class MilvusUploader(Uploader):
     upload_config: MilvusUploaderConfig
     connector_type: str = CONNECTOR_TYPE
 
+    @DestinationConnectionError.wrap
+    def precheck(self):
+        with self.get_client() as client:
+            if not client.has_collection(self.upload_config.collection_name):
```
[the remark this answers is not in the document]
Comment (ds-filipknefel) [reply unstructured_ingest/v2/processes/connectors/milvus.py:179]: Looking at the function docs and it's code I don't have a good reason to think that Milvus would provision a collection on [has_collection](https://milvus.io/api-reference/pymilvus/v2.4.x/MilvusClient/Collections/has_collection.md) call.

It sends a describe collection request and checks if it succeeds.

File: test/integration/connectors/test_milvus.py [hunk 2]
```diff
@@ -139,3 +143,26 @@ async def test_milvus_destination(
     uploader.run(path=staged_filepath, file_data=file_data)
     with uploader.get_client() as client:
         validate_count(client=client, expected_count=expected_count)
+
+
+@pytest.mark.tags(CONNECTOR_TYPE, DESTINATION_TAG)
+class TestPrecheck:
```
Comment (rbiseck3) [inline test/integration/connectors/test_milvus.py:149]: Why are we introducing a class based approach here?

Comment (rbiseck3) [reply test/integration/connectors/test_milvus.py:149]: Can we stick to single test methods to keep things consistent?

**Review by rbiseck3 — APPROVED** [review]
Small comment to fix but otherwise LGTM

File: test/integration/connectors/test_milvus.py (hunk 2 above)
Comment (mateuszkuprowski) [reply test/integration/connectors/test_milvus.py]: I've refactored this into two separate functions, good point!

**Pull request merged**